docs/adr/adr-006-befp.md

```diff
@@ -0,0 +1,118 @@
+# ADR #006: Celestia-Node Bad encoding fraud proof(BEFP)
+
+## Changelog
+
+- 2022.03.03 - init commit
+- 2022.03.08 - added pub-sub
+
+## Authors
+
+@vgonkivs @Bidon15 @adlerjohn @Wondertan
+
+## Context
+
+In case when a Full Node receives `ErrByzantineRow`/`ErrByzantineCol` from the [rsmt2d](https://github.com/celestiaorg/rsmt2d) library, it generates a fraud proof.  After generation of it(fraud proof), a Full Node is broadcasting the fraud proof to Light Nodes, so they(Light Nodes) are notified that a block could be malicious.
+
+## Decision
+
+Started disscussion within:
+
+- https://github.com/celestiaorg/celestia-node/issues/4
+- https://github.com/celestiaorg/celestia-node/issues/263
+
+## Detailed Design
+A fraud proof is generated if recovered data does not match with its respective row/column roots during block repair process
+
+The result of RepairExtendedDataSquare will be an error [ErrByzantineRow](https://github.com/celestiaorg/rsmt2d/blob/f34ec414859fc834835ea97ed54300404eec1ac5/extendeddatacrossword.go#L18-L22)/[ErrByzantineCol](https://github.com/celestiaorg/rsmt2d/blob/f34ec414859fc834835ea97ed54300404eec1ac5/extendeddatacrossword.go#L28-L32):
+
+- Both errors consist of 
+  - row/column numbers that do not match with the Merkle root
+  - shares that were successfully repaired and verified (all correct shares).
+
+`ErrByzantineRow`/`ErrByzantineCol` are returned from `RetrevieData`. In addition, `das.Daser`:
+
+1. Generates a MerkleProofs for respective verified shares
+2. Creates a BEFP
+3. Broadcasts it(BEFP) to light nodes via separate sub-service.
```

Comment:
```suggestion
3. Broadcasts the BEFP to light nodes via separate gossipsub network.
```
We're talking about `FraudSub` here, right? If so, we can say "Broadcasts the BEFP to the FraudSub gossipsub network which all Light Nodes listen to by default.